homeassistant/helpers/intent.py

```diff
@@ -1318,3 +1324,79 @@ def as_dict(self) -> dict[str, Any]:
         response_dict["data"] = response_data
 
         return response_dict
+
+
+class IntentTool(llm.Tool):
```

Comment:
this should live in `llm.py`. Intents should remain their own thing without knowing that LLMs consume them.

Reply:
This was my suggestion to flip around, given how complex it is with lots of intent specific logic. Are there ways to reduce the scope of what this is doing to let intent code handle all these lookups?

Reply:
Moved to `llm.py`. Also there is not much intent-specific logic left.